net: openthread: add microseconds timer API

edmont opened the pull request:
Add OpenThread API to handle microseconds timer, to be used initially by CSL receiver.

`CONFIG_OPENTHREAD_PLATFORM_USEC_TIMER_ENABLE` is removed and microseconds timer enabling is tied now to the functionalities that use it (currently CSL receiver)

The attached patch can be used to verify the correctness of the new API. It just substitutes the milliseconds timer API implementation by microseconds timer calls.

[timer_replace_patch.txt](https://github.com/zephyrproject-rtos/zephyr/files/6295784/timer_replace_patch.txt)


Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

rlubos commented on subsys/net/l2/openthread/Kconfig:
```diff
@@ -286,10 +286,4 @@ config OPENTHREAD_MAC_SOFTWARE_CSMA_BACKOFF_ENABLE
 	  Set y to enable software CSMA backoff. The option can be disabled if
 	  the radio has hardware support for this feature (IEEE802154_HW_CSMA).
 
-config OPENTHREAD_PLATFORM_USEC_TIMER_ENABLE
```
Do you recall what was the purpose of this config? Is it just that the option was added but not actually implemented on the Zephyr side?

edmont replied:
Exactly, it was added here https://github.com/zephyrproject-rtos/zephyr/pull/29366, but never used.

rlubos commented on subsys/net/lib/openthread/platform/openthread-core-zephyr-config.h:
```diff
@@ -98,11 +98,10 @@
-#ifdef CONFIG_OPENTHREAD_PLATFORM_USEC_TIMER_ENABLE
-#define OPENTHREAD_CONFIG_PLATFORM_USEC_TIMER_ENABLE 1
-#endif
+#define OPENTHREAD_CONFIG_PLATFORM_USEC_TIMER_ENABLE (OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE && \
+(OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2))
```
Just a small style nit
```suggestion
#define OPENTHREAD_CONFIG_PLATFORM_USEC_TIMER_ENABLE                           \
       (OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE &&                          \
        (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2))
 

```

edmont replied:
It seems this will make `checkpatch` fail with `WARNING:LEADING_SPACE`.

rlubos replied:
Sorry, I should've put tab in the sample code as an initial indent when copying. This is the format that `clang-format` suggests.

edmont replied:
Thanks, fixed.

rlubos reviewed (approved):
Looks good

rlubos commented:
Apologies for deleting the reviewers, that wasn't my intention (just some GH UI oddiness).

edmont committed:
net: openthread: add microseconds timer API

Add OpenThread API to handle microseconds timer, to be used initially
by CSL receiver.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>
